test/pkg/environment/common/default_nodepool.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: karpenter.sh/v1
```

Comment:
Do we absolutely need this default? Or can we control this until we get a request to have something different?

Reply:
Is the concern with injecting cloudprovider-specific requirements into the requirements section?

Reply:
Yeah, the concern is exactly that. We need to be able to allow cloudprovider to inject requirements. We could give providers a global variable to define their requirements, but I figured that would be very restrictive without much upside IMO